### PR TITLE
feat: Enhance Query Parameters

### PR DESCRIPTION
**Pull request by EnriqueL8**
Add the new filtering functionalities from KituraContracts `1.0.0`, such as:
```
GreaterThan
LowerThan
GreaterThanOrEqual
LowerThanOrEqual
InclusiveRange
ExclusiveRange
Ordering
Pagination
```

And prevent SQL Injections with QueryParameters in Select and Delete Queries

### CODE REVIEW

**Comment by ianpartridge** [comment]
@EnriqueL8 ^^^ test failures

File: Sources/SwiftKueryORM/Model.swift [hunk 1]
```diff
@@ -341,55 +341,83 @@ public extension Model {
...
       return
     }
 
-    let query = Select(from: table).where(filter)
-    Self.executeQuery(query: query, using: db, onCompletion)
+    var query: Select = Select(from: table)
+    var parameters: [Any?]? = nil
+    if let queryParams = queryParams {
+      do {
+        let resultTuple: (query: Select, parameters: [Any?]?) = try getSelectQueryWithFilters(query: query, queryParams: queryParams, table: table)
```
Comment (ianpartridge) [inline Sources/SwiftKueryORM/Model.swift]: I don't think you need the intermediate tuple, you can just do:

```swift
(query, parameters) = try getSelectQueryWithFilters(query: query, queryParams: queryParams, table: table)
```

File: Sources/SwiftKueryORM/Model.swift [hunk 2]
```diff
@@ -341,55 +341,83 @@ public extension Model {
...
```
Comment (ianpartridge) [inline Sources/SwiftKueryORM/Model.swift]: Same comment.

File: Sources/SwiftKueryORM/Model.swift [hunk 3]
```diff
@@ -341,55 +341,83 @@ public extension Model {
...
       return
     }
 
-    let query = Select(from: table).where(filter)
-    Self.executeQuery(query: query, using: db) { (tuples: [(I, Self)]?, error: RequestError?) in
+    var query: Select = Select(from: table)
+    var parameters: [Any?]? = nil
+    if let queryParams = queryParams {
+      do {
+        let resultTuple: (query: Select, parameters: [Any?]?) = try getSelectQueryWithFilters(query: query, queryParams: queryParams, table: table)
```
Comment (ianpartridge) [inline Sources/SwiftKueryORM/Model.swift]: Same comment.

File: Sources/SwiftKueryORM/Model.swift [hunk 4]
```diff
@@ -462,19 +490,38 @@ public extension Model {
...
     }
 
-    let query = Delete(from: table).where(filter)
-    Self.executeQuery(query: query, using: db, onCompletion)
+    var query: Delete = Delete(from: table)
+    var parameters: [Any?]? = nil
+    if let queryParams = queryParams {
+      do {
+        let values: [String: Any] = try QueryEncoder().encode(queryParams)
+        if values.count < 0 {
```
Comment (ianpartridge) [inline Sources/SwiftKueryORM/Model.swift]: How can this ever be less than zero?

File: Sources/SwiftKueryORM/Model.swift [hunk 5]
```diff
@@ -850,39 +914,186 @@ public extension Model {
     return try Database.tableInfo.getTable((Self.idColumnName, Self.idColumnType), Self.tableName, for: Self.self)
   }
 
+  /**
+    This functions accepts a Select query, an instance of QueryParams and the database table.
+    It returns the updated Select query containing the filtering values extracted from the QueryParameters and the parameters to inject in the SQL Query (this is to prevent SQL Injection)
+  */
+  private static func getSelectQueryWithFilters<Q: QueryParams>(query: Select, queryParams: Q, table: Table) throws -> (query: Select, parameters: [Any?]?) {
+      let values: [String: Any] = try QueryEncoder().encode(queryParams)
+      if values.count < 0 {
```
Comment (ianpartridge) [inline Sources/SwiftKueryORM/Model.swift]: Same comment.

File: Sources/SwiftKueryORM/Model.swift (hunk 1 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: Done, that is much nicer!

File: Sources/SwiftKueryORM/Model.swift (hunk 2 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: Done

File: Sources/SwiftKueryORM/Model.swift (hunk 4 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: If all the values in the QueryParams struct are set to nil, it will return `[:]`

Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: https://github.com/IBM-Swift/KituraContracts/blob/master/Sources/KituraContracts/CodableQuery/QueryEncoder.swift#L68

Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: https://github.com/IBM-Swift/KituraContracts/blob/master/Sources/KituraContracts/CodableQuery/QueryEncoder.swift#L139

Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: oh less than 1, my bad

Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: Done

File: Sources/SwiftKueryORM/Model.swift (hunk 5 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: Done

File: Sources/SwiftKueryORM/Model.swift (hunk 3 above)
Comment (EnriqueL8) [reply Sources/SwiftKueryORM/Model.swift]: Done

**Pull request merged**